### PR TITLE
Return ICMPv6 Destination Unreachable for unknown destinations

### LINKED ISSUE #180: can't ping if launch yggdrasil
I [haven't](url) ipv6 address from provider. I haven't any ipv6 address. But, if I launch yggdrassil, it's properly connected to the nodes and I can ping into yggdrasil network. But, I can't ping any clearnet ipv4 addresses. What I do wrong? How can I fix it?

### PR DESCRIPTION
**Pull request by neilalexander**
I have absolutely no idea if this will fix #180 and #437 but it's about the only thing that I can think of that might.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,6 +12,7 @@ require (
 	github.com/mitchellh/mapstructure v1.1.2
 	github.com/vishvananda/netlink v1.0.0
 	github.com/vishvananda/netns v0.0.0-20190625233234-7109fa855b0f // indirect
+	github.com/yggdrasil-network/yggdrasil-extras v0.0.0-20201206140615-845aca5e7894 // indirect
```
Comment (Arceliar) [inline go.mod:15]: Can you `go mod tidy` to see if this goes away? I see no reason why `yggdrasil-extras` should be included here (and it introduces a circular dependency), so I would expect this to not be needed.

File: go.sum [hunk 2]
```diff
@@ -31,6 +31,8 @@ github.com/vishvananda/netlink v1.0.0 h1:bqNY2lgheFIu1meHUFSH3d7vG93AFyqg3oGbJCO
 github.com/vishvananda/netlink v1.0.0/go.mod h1:+SR5DhBJrl6ZM7CoCKvpw5BKroDKQ+PJqOg65H/2ktk=
 github.com/vishvananda/netns v0.0.0-20190625233234-7109fa855b0f h1:nBX3nTcmxEtHSERBJaIo1Qa26VwRaopnZmfDQUXsF4I=
 github.com/vishvananda/netns v0.0.0-20190625233234-7109fa855b0f/go.mod h1:ZjcWmFBXmLKZu9Nxj3WKYEafiSqer2rnvPr0en9UNpI=
+github.com/yggdrasil-network/yggdrasil-extras v0.0.0-20201206140615-845aca5e7894 h1:CiLvZWLT38MlyrfxhVVsnOqVBL/b5EUpXiOQnqNPLU0=
+github.com/yggdrasil-network/yggdrasil-extras v0.0.0-20201206140615-845aca5e7894/go.mod h1:xQdsh08Io6nV4WRnOVTe6gI8/2iTvfLDQ0CYa5aMt+I=
```
Comment (Arceliar) [inline go.sum:35]: Same as the `go.mod` comment, I don't see any reason why this should be needed...

**Review by Arceliar — COMMENTED** [review]
It looks like some unused dependencies may have found their way into `go.mod` and `go.sum`, which I'd like to have cleaned up before this is merged. I haven't tested it, but in general I think it makes sense to return an error such as this instead of silently dropping the packet. Whether or not applications or the OS will do the right thing is another matter... I could imagine some weird interactions may occur if CKR settings are modified after startup, so we may want to regression test this one a bit before merging it.

File: go.mod (hunk 1 above)
Comment (neilalexander) [reply go.mod]: I expect this has happened because I've done `./build -i` and it's been pulled in. `go mod tidy` has removed it.

**Review by Arceliar — APPROVED** [review]
Looks good to me, though I wouldn't be opposed to more testing (mostly changing CKR configuration at runtime) before merging into develop, just in case it does something unexpected.

**Comment by neilalexander** [comment]
I've done some testing this evening and it seems that adding and removing CKR routes at runtime works fine with this—the ICMPv6 responses are sent when the CKR route is removed and they aren't when it is re-added.

Linux doesn't appear to cache the unreachability anywhere, presumably because it is a layer 3 tunnel and there's no Neighbour Discovery record in memory for the result to be cached in. Therefore pinging immediately after making changes seems to do the right thing.

**Pull request merged**